packages/arb-token-bridge-ui/src/components/TransactionHistory/TransactionsTableDetails.tsx

```diff
@@ -0,0 +1,243 @@
+import { XMarkIcon } from '@heroicons/react/24/outline'
+import { ArrowRightIcon } from '@heroicons/react/24/solid'
+import { Fragment, PropsWithChildren, useMemo } from 'react'
+import { Dialog, Transition } from '@headlessui/react'
+import Image from 'next/image'
+import dayjs from 'dayjs'
+import CctpLogoColor from '@/images/CctpLogoColor.svg'
+import ArbitrumLogo from '@/images/ArbitrumLogo.svg'
+
+import { useTxDetailsStore } from './TransactionHistory'
+import { getExplorerUrl, getNetworkName, isNetwork } from '../../util/networks'
+import { NetworkImage } from '../common/NetworkImage'
+import { TransactionsTableTokenImage } from './TransactionsTableTokenImage'
+import { formatAmount, formatUSD } from '../../util/NumberUtils'
+import { useETHPrice } from '../../hooks/useETHPrice'
+import { ExternalLink } from '../common/ExternalLink'
+import { TransactionsTableDetailsSteps } from './TransactionsTableDetailsSteps'
+import { Button } from '../common/Button'
+import { GET_HELP_LINK } from '../../constants'
+import { useTransactionHistory } from '../../hooks/useTransactionHistory'
+import { shortenAddress } from '../../util/CommonUtils'
+import { useNativeCurrency } from '../../hooks/useNativeCurrency'
+import { getProvider, isTxCompleted } from './helpers'
+import { AssetType } from '../../hooks/arbTokenBridge.types'
+import { Address } from '../../util/AddressUtils'
+
+const DetailsBox = ({
+  children,
+  header
+}: PropsWithChildren<{ header?: string }>) => {
+  return (
+    <div className="my-2 flex w-full flex-col rounded border border-white/30 bg-black p-3 font-light text-white">
+      {header && (
+        <h4 className="mb-2 text-xs uppercase text-white/60">{header}</h4>
+      )}
+      {children}
+    </div>
+  )
+}
+
+export const TransactionsTableDetails = ({
+  address
+}: {
+  address: Address | undefined
+}) => {
+  const { tx: txFromStore, isOpen, close, reset } = useTxDetailsStore()
+  const { ethToUSD } = useETHPrice()
+  const { transactions } = useTransactionHistory(address)
+
+  const tx = useMemo(() => {
+    if (!txFromStore) {
+      return null
+    }
+
+    // we need to get tx from the hook to make sure we have up to date details, e.g. status
+    return transactions.find(
+      t =>
+        t.parentChainId === txFromStore.parentChainId &&
+        t.childChainId === txFromStore.childChainId &&
+        t.txId === txFromStore.txId
+    )
+  }, [transactions, txFromStore])
+
+  const nativeCurrency = useNativeCurrency({
+    provider: getProvider(tx?.parentChainId ?? 1)
+  })
+
+  if (!tx || !address) {
+    return null
+  }
+
+  const showPriceInUsd =
+    isNetwork(tx.parentChainId).isEthereumMainnet &&
```

Comment:
But we can also show price when it's a deposit to an L3, as long as the currency is ETH, so we should just check if the network is mainnet, not ethereum mainnet

Reply:
updated